### PR TITLE
dist/tools: Improve Headerguard Check

### PR DESCRIPTION
**Pull request by crasbe**
### Contribution description

The headerguard check did not catch the case when a `#pragma once` is present in the file followed by classic header guards.
Apparently I did not anticipate this corner case in #21367...

This commit also improves the error messages to give a better reason why the test failed.

#### Improved Error Messages: Missing Header Guards

Remove the header guards from a header of your liking, for example `boards/feather-m0/include/board.h`

Current Master:
```
cbuec@W11nMate:~/RIOTstuff/riot-gettingstarted/RIOT$ ./dist/tools/headerguards/check.sh
boards/feather-m0/include/board.h: no / broken header guard
```

This PR:
```
cbuec@W11nMate:~/RIOTstuff/riot-gettingstarted/RIOT$ git diff boards/feather-m0/include/board.h
diff --git a/boards/feather-m0/include/board.h b/boards/feather-m0/include/board.h
index bd410fdfb2..2ff00cf26b 100644
--- a/boards/feather-m0/include/board.h
+++ b/boards/feather-m0/include/board.h
@@ -17,9 +17,6 @@
  * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
  */

-#ifndef BOARD_H
-#define BOARD_H
-
 #include "cpu.h"
 #include "periph_conf.h"
 #include "periph_cpu.h"
cbuec@W11nMate:~/RIOTstuff/riot-gettingstarted/RIOT$ ./dist/tools/headerguards/check.sh
boards/feather-m0/include/board.h: no header guards found
```

#### Corner Case: Pragma Once *and* Headerguards

Add a ´#pragma once´ to a header of your liking that still has the old format, for example `boards/feather-m0/include/board.h`

Current Master (PASS, but should not pass):
```
cbuec@W11nMate:~/RIOTstuff/riot-gettingstarted/RIOT$ git diff boards/feather-m0/include/board.h
diff --git a/boards/feather-m0/include/board.h b/boards/feather-m0/include/board.h
index bd410fdfb2..c0ce0440b4 100644
--- a/boards/feather-m0/include/board.h
+++ b/boards/feather-m0/include/board.h
@@ -6,6 +6,8 @@
  * directory for more details.
  */

+#pragma once
+
 /**
  * @ingroup     boards_feather-m0
  *
cbuec@W11nMate:~/RIOTstuff/riot-gettingstarted/RIOT$ ./dist/tools/headerguards/check.sh
cbuec@W11nMate:~/RIOTstuff/riot-gettingstarted/RIOT$
```

With this PR:
```
cbuec@W11nMate:~/RIOTstuff/riot-gettingstarted/RIOT$ ./dist/tools/headerguards/check.sh
boards/feather-m0/include/board.h: #pragma once and classic header guards used in the same file
```

### Issues/PRs references

`#pragma once` check introduced in #21367.

Discovered in https://github.com/RIOT-OS/RIOT/pull/21479#discussion_r2084905034

### CODE REVIEW

File: dist/tools/headerguards/headerguards.py [hunk 1]
```diff
@@ -59,6 +59,13 @@ def fix_headerguard(filename):
                 guard_found += 1
                 guard_name = line[8:].rstrip()
                 line = "#ifndef %s\n" % (supposed)
+        elif guard_found == 0 and pragma_once_found != 0:
```
[the remark this answers is not in the document]
Comment (crasbe) [reply dist/tools/headerguards/headerguards.py:62]: I don't think I could achieve that goal without duplicating the line suggestion code though.

[the remark this answers is not in the document]
Comment (crasbe) [reply dist/tools/headerguards/headerguards.py:61]: The fixup added some documentation about the valid cases. For the error cases, the warning message is probably clear enough about what went wrong.

[the remark this answers is not in the document]
Comment (mguetschow) [reply dist/tools/headerguards/headerguards.py:61]: Hum, not sure I understand. I meant: 

```py
        if guard_found == 0:
            if pragma_once_found == 0 and line.startswith("#ifndef"):
                guard_found += 1
                guard_name = line[8:].rstrip()
                line = "#ifndef %s\n" % (supposed)
            elif pragma_once_found != 0 and line.startswith("#ifndef %s" % supposed):
                # check if there is pragma once *and* headerguards, but avoid false
                # positives by narrowing in on the to-be-expected guard
                guard_found += 1
                guard_name = line[8:].rstrip()
                line = ""
```

[the remark this answers is not in the document]
Comment (crasbe) [reply dist/tools/headerguards/headerguards.py]: Oh, right. Yeah, that would work.

**Review by mguetschow — APPROVED** [review]
LGTM then, please squash!

**Comment by crasbe** [comment]
Thanks for reviewing :)
